src/publish.js

```diff
@@ -114,28 +114,32 @@ have published it.\
       };
 
       return new Promise((resolve, _reject) => {
-        const requestTags = () => void request.get(requestSettings, (_error, response, tags) => {
-          tags ??= [];
-          if (response?.statusCode === 200) {
-            if (tags.find(elem => elem.name === tag) != null) {
-              resolve();
-              return;
+        const requestTags = () => {
+          request.get(
+            requestSettings,
+            (_error, response, tags) => {
+              tags ??= [];
+              if (response?.statusCode === 200) {
+                if (tags.some(t => t.name === tag)) {
+                  resolve();
+                  return;
+                }
+              }
+              if (--retryCount <= 0) {
+                return void resolve();
+              }
             }
-          }
-          if (--retryCount <= 0) {
-            return void resolve();
-          }
-
-          setTimeout(requestTags, interval);
-        });
+          )
+        };
+        setTimeout(requestTags, interval);
       });
     }
```

Comment:
I rewrote this section because the original code (in my opinion) went out of its way to be inscrutable.

An arrow function can skip surrounding braces if the entire body is a single statement. That doesn't mean it's a good idea to treat a complex multi-line statement as though it were a one-liner. I think that this is one of the reasons why the author ended up putting the `setTimeout` on the inside of the `requestTags` callback instead of the outside.

Reply:
I just realized you're right that I shouldn't have moved the `setTimeout` — I should've just invoked `requestTags` once outside the block. I'll make that change.

I'm still strongly in favor of reformatting that code; I'd prefer to do it now rather than in a separate PR.

Reply:
And you're right that this code path still isn't hit in the tests — even in the new tests I wrote. I mocked anything that talked to the outside world. I don't know what else to do for a method whose only purpose is to wait until a remote server responds a certain way.

Reply:
Re: **tests**, yeah, our tests are already not platinum-tier 110% complete, and I don't know off the top of my head how to mock an error code response from the hypothetical server. Certainly not going to ask for this, it's not obvious to me how doable it would be.

As for readability, yeah the original syntax was a little... weird. It's still kinda gnarly due to being so nested and having so many different JS features used in such a short span. **I think you're right that a refactor is warranted for readability.**

**I ended up with this locally just so I could read and understand what was happening.**

<details><summary>Code snippet (click to expand):</summary>

```js
    waitForTagToBeAvailable(pack, tag) {
      let retryCount = 5;
      const interval = 1000;
      const requestSettings = {
        url: `https://api.github.com/repos/${Packages.getRepository(pack)}/tags`,
        json: true
      };

      return new Promise((resolve, _reject) => {
        const requestTags = () => void request.get(requestSettings, (_error, response, tags) => {
          tags ??= [];
          if (response?.statusCode === 200) {
            if (tags.find(elem => elem.name === tag) != null) {
              resolve();
              return;
            }
          }
          if (--retryCount <= 0) {
            return void resolve();
          }

          setTimeout(requestTags, interval);
        });
      });
    }
```

</details>

Took me a lot of staring at it and having Pulsar highlight matching brackets/parentheses for me and trying to run it to see which line errored...